### ISSUE TITLE
Error setting up database on fresh install

### ISSUE THREAD

**Issue by rEnr3n** [wallabag/wallabag#3739]
:warning: If your issue is about an error during fetching a link, please read: http://doc.wallabag.org/en/user/errors_during_fetching.html#how-can-i-help-to-fix-that

### Issue details

Please provide issue details here.
Remember, this is _not_ a place to ask questions. For that, go to https://community.wallabag.org/ (forum) or http://gitter.im/wallabag/wallabag (chat).

### Environment

* wallabag version (or git revision) that exhibits the issue: 2.3.3
* How did you install wallabag? Via `git clone` or by downloading the package? https://aur.archlinux.org/packages/wallabag/
* Last wallabag version that did not exhibit the issue (if applicable): FRESH INSTALL
* php version: 7.2.11
* OS: ArchLinux ARM
* type of hosting (shared or dedicated): dedicated (home server)
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): PostgreSQL

### Steps to reproduce/test case

Please provide necessary steps for reproduction of this issue, or better the
reduced test case (without any external dependencies, if possible).

1. Install wallabag from AUR using any aur helper (e.g. pikaur -S wallabag)
2. Configure parameters.yml
```
# This file is auto-generated during the composer install
parameters:
    database_driver: pdo_pgsql
    database_driver_class: Wallabag\CoreBundle\Doctrine\DBAL\Driver\CustomPostgreSQLDriver
    database_host: 127.0.0.1
    database_port: null
    database_name: wallabag
    database_user: wb
    database_password: null
    database_path: null
    database_table_prefix: wallabag_
    database_socket: null
    database_charset: utf8
    domain_name: 'https://wallabag.server.lan'
    mailer_transport: smtp
    mailer_host: 127.0.0.1
    mailer_user: null
    mailer_password: null
    locale: en
    secret: <REDACTED>
    twofactor_auth: false
    twofactor_sender: no-reply@wallabag.org
    fosuser_registration: true
    fosuser_confirmation: true
    from_email: no-reply@wallabag.org
    rss_limit: 50
    rabbitmq_host: localhost
    rabbitmq_port: 5672
    rabbitmq_user: guest
    rabbitmq_password: guest
    rabbitmq_prefetch_count: 10
    redis_scheme: unix
    redis_host: localhost
    redis_port: null
    redis_path: null
    redis_password: null
```
3. Run this command as web user `php /usr/share/webapps/wallabag/bin/console wallabag:install --env=prod`

I get this error during step 2.
<details>
<summary>Error log</summary>

```
Step 2 of 4: Setting up database.
---------------------------------

 It appears that your database already exists. Would you like to reset it? (yes/no) [no]:
 > yes

 Dropping database, creating database and schema...

                                                                                                                                                     
  [RuntimeException]                                                                                                                                 
  The command "doctrine:migrations:migrate" generates some errors:                                                                                   
                                                                                                                                                     
                                                                                                                                                     
                      Application Migrations                                                                                                         
                                                                                                                                                     
                                                                                                                                                     
  Migrating up to 20171125164500 from 0                                                                                                              
                                                                                                                                                     
    ++ migrating 20160401000000                                                                                                                      
                                                                                                                                                     
       -> CREATE TABLE "wallabag_craue_config_setting" (name VARCHAR(255) NOT NULL, value VARCHAR(255) DEFAULT NULL, section VARCHAR(255) DEFAULT N  
  ULL, PRIMARY KEY(name));                                                                                                                           
       -> CREATE UNIQUE INDEX UNIQ_5D9649505E237E06 ON "wallabag_craue_config_setting" (name);                                                       
       -> CREATE TABLE "wallabag_entry" (id INT NOT NULL, user_id INT DEFAULT NULL, title TEXT DEFAULT NULL, url TEXT DEFAULT NULL, is_archived BOO  
  LEAN NOT NULL, is_starred BOOLEAN NOT NULL, content TEXT DEFAULT NULL, created_at TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL, updated_at TIMESTAMP(0  
  ) WITHOUT TIME ZONE NOT NULL, mimetype TEXT DEFAULT NULL, language TEXT DEFAULT NULL, reading_time INT DEFAULT NULL, domain_name TEXT DEFAULT NUL  
  L, preview_picture TEXT DEFAULT NULL, is_public BOOLEAN DEFAULT 'false', PRIMARY KEY(id));                                                         
       -> CREATE INDEX IDX_F4D18282A76ED395 ON "wallabag_entry" (user_id);                                                                           
       -> CREATE TABLE "wallabag_entry_tag" (entry_id INT NOT NULL, tag_id INT NOT NULL, PRIMARY KEY(entry_id, tag_id));                             
       -> CREATE INDEX IDX_C9F0DD7CBA364942 ON "wallabag_entry_tag" (entry_id);                                                                      
       -> CREATE INDEX IDX_C9F0DD7CBAD26311 ON "wallabag_entry_tag" (tag_id);                                                                        
       -> CREATE TABLE "wallabag_config" (id INT NOT NULL, user_id INT DEFAULT NULL, theme VARCHAR(255) NOT NULL, items_per_page INT NOT NULL, lang  
  uage VARCHAR(255) NOT NULL, rss_token VARCHAR(255) DEFAULT NULL, rss_limit INT DEFAULT NULL, reading_speed DOUBLE PRECISION DEFAULT NULL, PRIMARY  
   KEY(id));                                                                                                                                         
       -> CREATE UNIQUE INDEX UNIQ_87E64C53A76ED395 ON "wallabag_config" (user_id);                                                                  
       -> CREATE TABLE "wallabag_tagging_rule" (id INT NOT NULL, config_id INT DEFAULT NULL, rule VARCHAR(255) NOT NULL, tags TEXT NOT NULL, PRIMAR  
  Y KEY(id));                                                                                                                                        
       -> CREATE INDEX IDX_2D9B3C5424DB0683 ON "wallabag_tagging_rule" (config_id);                                                                  
       -> COMMENT ON COLUMN "wallabag_tagging_rule".tags IS '(DC2Type:simple_array)';                                                                
       -> CREATE TABLE "wallabag_tag" (id INT NOT NULL, label TEXT NOT NULL, slug VARCHAR(128) NOT NULL, PRIMARY KEY(id));                           
       -> CREATE UNIQUE INDEX UNIQ_4CA58A8C989D9B62 ON "wallabag_tag" (slug);                                                                        
       -> CREATE TABLE "wallabag_oauth2_clients" (id INT NOT NULL, random_id VARCHAR(255) NOT NULL, redirect_uris TEXT NOT NULL, secret VARCHAR(255  
  ) NOT NULL, allowed_grant_types TEXT NOT NULL, PRIMARY KEY(id));                                                                                   
       -> COMMENT ON COLUMN "wallabag_oauth2_clients".redirect_uris IS '(DC2Type:array)';                                                            
       -> COMMENT ON COLUMN "wallabag_oauth2_clients".allowed_grant_types IS '(DC2Type:array)';                                                      
       -> CREATE TABLE "wallabag_oauth2_access_tokens" (id INT NOT NULL, client_id INT NOT NULL, user_id INT DEFAULT NULL, token VARCHAR(255) NOT N  
  ULL, expires_at INT DEFAULT NULL, scope VARCHAR(255) DEFAULT NULL, PRIMARY KEY(id));                                                               
       -> CREATE UNIQUE INDEX UNIQ_368A42095F37A13B ON "wallabag_oauth2_access_tokens" (token);                                                      
       -> CREATE INDEX IDX_368A420919EB6921 ON "wallabag_oauth2_access_tokens" (client_id);                                                          
       -> CREATE INDEX IDX_368A4209A76ED395 ON "wallabag_oauth2_access_tokens" (user_id);                                                            
       -> CREATE TABLE "wallabag_oauth2_refresh_tokens" (id INT NOT NULL, client_id INT NOT NULL, user_id INT DEFAULT NULL, token VARCHAR(255) NOT   
  NULL, expires_at INT DEFAULT NULL, scope VARCHAR(255) DEFAULT NULL, PRIMARY KEY(id));                                                              
       -> CREATE UNIQUE INDEX UNIQ_20C9FB245F37A13B ON "wallabag_oauth2_refresh_tokens" (token);                                                     
       -> CREATE INDEX IDX_20C9FB2419EB6921 ON "wallabag_oauth2_refresh_tokens" (client_id);                                                         
       -> CREATE INDEX IDX_20C9FB24A76ED395 ON "wallabag_oauth2_refresh_tokens" (user_id);                                                           
       -> CREATE TABLE "wallabag_oauth2_auth_codes" (id INT NOT NULL, client_id INT NOT NULL, user_id INT DEFAULT NULL, token VARCHAR(255) NOT NULL  
  , redirect_uri TEXT NOT NULL, expires_at INT DEFAULT NULL, scope VARCHAR(255) DEFAULT NULL, PRIMARY KEY(id));                                      
       -> CREATE UNIQUE INDEX UNIQ_EE52E3FA5F37A13B ON "wallabag_oauth2_auth_codes" (token);                                                         
       -> CREATE INDEX IDX_EE52E3FA19EB6921 ON "wallabag_oauth2_auth_codes" (client_id);                                                             
       -> CREATE INDEX IDX_EE52E3FAA76ED395 ON "wallabag_oauth2_auth_codes" (user_id);                                                               
       -> CREATE TABLE "wallabag_user" (id INT NOT NULL, username VARCHAR(180) NOT NULL, username_canonical VARCHAR(180) NOT NULL, email VARCHAR(18  
  0) NOT NULL, email_canonical VARCHAR(180) NOT NULL, enabled BOOLEAN NOT NULL, salt VARCHAR(255) NOT NULL, password VARCHAR(255) NOT NULL, last_lo  
  gin TIMESTAMP(0) WITHOUT TIME ZONE DEFAULT NULL, locked BOOLEAN NOT NULL, expired BOOLEAN NOT NULL, expires_at TIMESTAMP(0) WITHOUT TIME ZONE DEF  
  AULT NULL, confirmation_token VARCHAR(255) DEFAULT NULL, password_requested_at TIMESTAMP(0) WITHOUT TIME ZONE DEFAULT NULL, roles TEXT NOT NULL,   
  credentials_expired BOOLEAN NOT NULL, credentials_expire_at TIMESTAMP(0) WITHOUT TIME ZONE DEFAULT NULL, name TEXT DEFAULT NULL, created_at TIMES  
  TAMP(0) WITHOUT TIME ZONE NOT NULL, updated_at TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL, authCode INT DEFAULT NULL, twoFactorAuthentication BOOLEA  
  N NOT NULL, trusted TEXT DEFAULT NULL, PRIMARY KEY(id));                                                                                           
       -> CREATE UNIQUE INDEX UNIQ_1D63E7E592FC23A8 ON "wallabag_user" (username_canonical);                                                         
       -> CREATE UNIQUE INDEX UNIQ_1D63E7E5A0D96FBF ON "wallabag_user" (email_canonical);                                                            
       -> CREATE UNIQUE INDEX UNIQ_1D63E7E5C05FB297 ON "wallabag_user" (confirmation_token);                                                         
       -> COMMENT ON COLUMN "wallabag_user".roles IS '(DC2Type:array)';                                                                              
       -> COMMENT ON COLUMN "wallabag_user".trusted IS '(DC2Type:json_array)';                                                                       
       -> CREATE TABLE "wallabag_annotation" (id INT NOT NULL, user_id INT DEFAULT NULL, entry_id INT DEFAULT NULL, text TEXT NOT NULL, created_at   
  TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL, updated_at TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL, quote VARCHAR(255) NOT NULL, ranges TEXT NOT NULL, P  
  RIMARY KEY(id));                                                                                                                                   
       -> CREATE INDEX IDX_A7AED006A76ED395 ON "wallabag_annotation" (user_id);                                                                      
       -> CREATE INDEX IDX_A7AED006BA364942 ON "wallabag_annotation" (entry_id);                                                                     
       -> COMMENT ON COLUMN "wallabag_annotation".ranges IS '(DC2Type:array)';                                                                       
       -> CREATE SEQUENCE "entry_id_seq" INCREMENT BY 1 MINVALUE 1 START 1;                                                                          
       -> CREATE SEQUENCE "config_id_seq" INCREMENT BY 1 MINVALUE 1 START 1;                                                                         
       -> CREATE SEQUENCE "tagging_rule_id_seq" INCREMENT BY 1 MINVALUE 1 START 1;                                                                   
       -> CREATE SEQUENCE "tag_id_seq" INCREMENT BY 1 MINVALUE 1 START 1;                                                                            
       -> CREATE SEQUENCE oauth2_clients_id_seq INCREMENT BY 1 MINVALUE 1 START 1;                                                                   
       -> CREATE SEQUENCE oauth2_access_tokens_id_seq INCREMENT BY 1 MINVALUE 1 START 1;                                                             
       -> CREATE SEQUENCE oauth2_refresh_tokens_id_seq INCREMENT BY 1 MINVALUE 1 START 1;                                                            
       -> CREATE SEQUENCE oauth2_auth_codes_id_seq INCREMENT BY 1 MINVALUE 1 START 1;                                                                
       -> CREATE SEQUENCE "user_id_seq" INCREMENT BY 1 MINVALUE 1 START 1;                                                                           
       -> CREATE SEQUENCE annotation_id_seq INCREMENT BY 1 MINVALUE 1 START 1;                                                                       
       -> ALTER TABLE "wallabag_entry" ADD CONSTRAINT FK_F4D18282A76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) NOT DEFERRABLE INIT  
  IALLY IMMEDIATE;                                                                                                                                   
       -> ALTER TABLE "wallabag_entry_tag" ADD CONSTRAINT FK_C9F0DD7CBA364942 FOREIGN KEY (entry_id) REFERENCES "wallabag_entry" (id) NOT DEFERRABL  
  E INITIALLY IMMEDIATE;                                                                                                                             
       -> ALTER TABLE "wallabag_entry_tag" ADD CONSTRAINT FK_C9F0DD7CBAD26311 FOREIGN KEY (tag_id) REFERENCES "wallabag_tag" (id) NOT DEFERRABLE IN  
  ITIALLY IMMEDIATE;                                                                                                                                 
       -> ALTER TABLE "wallabag_config" ADD CONSTRAINT FK_87E64C53A76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) NOT DEFERRABLE INI  
  TIALLY IMMEDIATE;                                                                                                                                  
       -> ALTER TABLE "wallabag_tagging_rule" ADD CONSTRAINT FK_2D9B3C5424DB0683 FOREIGN KEY (config_id) REFERENCES "wallabag_config" (id) NOT DEFE  
  RRABLE INITIALLY IMMEDIATE;                                                                                                                        
       -> ALTER TABLE "wallabag_oauth2_access_tokens" ADD CONSTRAINT FK_368A420919EB6921 FOREIGN KEY (client_id) REFERENCES "wallabag_oauth2_client  
  s" (id) NOT DEFERRABLE INITIALLY IMMEDIATE;                                                                                                        
       -> ALTER TABLE "wallabag_oauth2_access_tokens" ADD CONSTRAINT FK_368A4209A76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) NOT   
  DEFERRABLE INITIALLY IMMEDIATE;                                                                                                                    
       -> ALTER TABLE "wallabag_oauth2_refresh_tokens" ADD CONSTRAINT FK_20C9FB2419EB6921 FOREIGN KEY (client_id) REFERENCES "wallabag_oauth2_clien  
  ts" (id) NOT DEFERRABLE INITIALLY IMMEDIATE;                                                                                                       
       -> ALTER TABLE "wallabag_oauth2_refresh_tokens" ADD CONSTRAINT FK_20C9FB24A76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) NOT  
   DEFERRABLE INITIALLY IMMEDIATE;                                                                                                                   
       -> ALTER TABLE "wallabag_oauth2_auth_codes" ADD CONSTRAINT FK_EE52E3FA19EB6921 FOREIGN KEY (client_id) REFERENCES "wallabag_oauth2_clients"   
  (id) NOT DEFERRABLE INITIALLY IMMEDIATE;                                                                                                           
       -> ALTER TABLE "wallabag_oauth2_auth_codes" ADD CONSTRAINT FK_EE52E3FAA76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) NOT DEF  
  ERRABLE INITIALLY IMMEDIATE;                                                                                                                       
       -> ALTER TABLE "wallabag_annotation" ADD CONSTRAINT FK_A7AED006A76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) NOT DEFERRABLE  
   INITIALLY IMMEDIATE;                                                                                                                              
       -> ALTER TABLE "wallabag_annotation" ADD CONSTRAINT FK_A7AED006BA364942 FOREIGN KEY (entry_id) REFERENCES "wallabag_entry" (id) NOT DEFERRAB  
  LE INITIALLY IMMEDIATE;                                                                                                                            
                                                                                                                                                     
    ++ migrated (5.68s)                                                                                                                              
                                                                                                                                                     
    ++ migrating 20160410190541                                                                                                                      
                                                                                                                                                     
  Migration 20160410190541 failed during Execution. Error An exception occurred while executing 'SELECT min_value, increment_by FROM "entry_id_seq"  
  ':                                                                                                                                                 
                                                                                                                                                     
  SQLSTATE[42703]: Undefined column: 7 ERROR:  column "min_value" does not exist                                                                     
  LINE 1: SELECT min_value, increment_by FROM "entry_id_seq"                                                                                         
                 ^                                                                                                                                   
  17:11:03 ERROR     [console] Error thrown while running command "doctrine:migrations:migrate --no-interaction=1 --no-debug=1 --env=prod". Message  
  : "An exception occurred while executing 'SELECT min_value, increment_by FROM "entry_id_seq"':                                                     
                                                                                                                                                     
  SQLSTATE[42703]: Undefined column: 7 ERROR:  column "min_value" does not exist                                                                     
  LINE 1: SELECT min_value, increment_by FROM "entry_id_seq"                                                                                         
                 ^" ["error" => Doctrine\DBAL\Exception\InvalidFieldNameException { …},"command" => "doctrine:migrations:migrate --no-interaction=1  
   --no-debug=1 --env=prod","message" => """  An exception occurred while executing 'SELECT min_value, increment_by FROM "entry_id_seq"':\n  \n  SQ  
  LSTATE[42703]: Undefined column: 7 ERROR:  column "min_value" does not exist\n  LINE 1: SELECT min_value, increment_by FROM "entry_id_seq"\n       
              ^  """] []                                                                                                                             
                                                                                                                                                     
                                                                                                                                                     
    [Doctrine\DBAL\Exception\InvalidFieldNameException]                                                                                              
    An exception occurred while executing 'SELECT min_value, increment_by FROM "entry_id_seq"':                                                      
                                                                                                                                                     
    SQLSTATE[42703]: Undefined column: 7 ERROR:  column "min_value" does not exist                                                                   
    LINE 1: SELECT min_value, increment_by FROM "entry_id_seq"                                                                                       
                   ^                                                                                                                                 
                                                                                                                                                     
                                                                                                                                                     
                                                                                                                                                     
    [Doctrine\DBAL\Driver\PDOException]                                                                                                              
    SQLSTATE[42703]: Undefined column: 7 ERROR:  column "min_value" does not exist                                                                   
    LINE 1: SELECT min_value, increment_by FROM "entry_id_seq"                                                                                       
                   ^                                                                                                                                 
                                                                                                                                                     
                                                                                                                                                     
                                                                                                                                                     
    [PDOException]                                                                                                                                   
    SQLSTATE[42703]: Undefined column: 7 ERROR:  column "min_value" does not exist                                                                   
    LINE 1: SELECT min_value, increment_by FROM "entry_id_seq"                                                                                       
                   ^                                                                                                                                 
                                                                                                                                                     
                                                                                                                                                     
  doctrine:migrations:migrate [--write-sql] [--dry-run] [--query-time] [--allow-no-migration] [--configuration [CONFIGURATION]] [--db-configuration  
   [DB-CONFIGURATION]] [--db DB] [--em EM] [--shard SHARD] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|-  
  -no-interaction] [-e|--env ENV] [--no-debug] [--] <command> [<version>]
```
</details>

**Comment by j0k3r** [comment]
Which version of PG are you using?

**Comment by rEnr3n** [comment]
> Which version of PG are you using?

10.5

**Comment by j0k3r** [comment]
Travis is using the 9.6.6.
As far as I remember we have tested it under 10.1 but not further. Could you try on 10.1 to check if you still have the error?

**Comment by rEnr3n** [comment]
I'm still getting the error on 10.1. Although it did work on 9.6.5.

**Comment by Kdecherf** [comment]
This issue is related to Doctrine and was fixed in doctrine/dbal 2.7: https://github.com/doctrine/dbal/pull/2893

Sadly, 2.7 is not currently available on packagist.

**Comment by rEnr3n** [comment]
> Sadly, 2.7 is not currently available on packagist.

What do you mean? It looks updated here https://packagist.org/packages/doctrine/dbal

**Comment by Kdecherf** [comment]
Hm, I misinterpreted a dependency of doctrine/orm. I'll check again

**Comment by j0k3r** [comment]
Postgres 10 support is done since few release https://github.com/wallabag/wallabag/pull/3488
dbal version in wallabag is locked at ^2.5.12 because of `doctrine/doctrine-bundle`: https://packagist.org/packages/doctrine/doctrine-bundle#1.9.1

Also, dbal 2.6.0 required php 7.1 and wallabag still support php5.

**Comment by Kdecherf** [comment]
Beside the fact that dbal 2.6.0 requires php ^7.1, the lock to `^2.5.12` should allow updates up to <3.0.

**Comment by j0k3r** [comment]
As we force the php version in `composer.json` composer doesn't install 2.6.0 because it's incompatible with the forced php version

https://github.com/wallabag/wallabag/blob/ad77c3451f2c436484f6de033fe18c844dfc9267/composer.json#L132-L137

**Comment by Kdecherf** [comment]
EOL of PHP 5.6 and PHP 7.0 is set for December 2018, maybe we could deprecate them and bump the minimal version to 7.1 in the following months.

**Comment by j0k3r** [comment]
This is planned for wallabag 2.4.0

**Comment by Kdecherf** [comment]
ACK

**Comment by nebulade** [comment]
I am getting the same error, on php7.2 and also postgresql 10. I might misread this, but if I change the composer.json to not require php 5.6.0 but 7.2, the same error occurs. Is there something else I have to do?

**Comment by j0k3r** [comment]
There are a lot of other things to do https://github.com/wallabag/wallabag/pull/3758
Wallabag isn't ready for postgresql 10 and php 7.2 for now.

**Comment by rEnr3n** [comment]
Postgresql 11 has been released a while ago. Do I have to open another issue for that? The OS I'm using is a rolling release. So it should hit the repos soon.

**Comment by Kdecherf** [comment]
@rEnr3n if I understand correctly initial support of Pg 11 was added in dbal 2.8.0. I think we will have updated to this version by the release of wallabag 2.4.0, so no need to create another issue ;-)

**Comment by j0k3r** [comment]
Ok I found the bug.
We force the database version to 5.6 since https://github.com/wallabag/wallabag/pull/1903
When forcing the database version it applies to all possible database, MySQL, Postgres & SQLite. Which means, when we are using dbal, as we force the database version to 5.6, dbal think we are using Postgresql 5.6.0 and load the default platform `PostgreSqlPlatform` instead of `PostgreSQL100Platform` in your case. As this sequence stuff changed in Postgres 10 and not in Postgres 9 we now got that error.

Pfiou.
I'm going to check that but my guess is it won't be fixed for 2.3.x but in 2.4.0.
